### PR TITLE
Add Wash config file

### PR DESCRIPTION
**Pull request by ekinanp**
Resolves https://github.com/puppetlabs/wash/issues/132

Contributions to this project require sign-off consistent with the [Developers Certificate of Origin](https://developercertificate.org). This can be as simple as using `git commit -s` on each commit.

### CODE REVIEW

File: cmd/internal/server/core.go [hunk 1]
```diff
@@ -26,10 +24,10 @@ import (
 // Opts exposes additional configuration for server operation.
 type Opts struct {
 	CPUProfilePath      string
-	ExternalPluginsPath string
+	ExternalPlugins     []plugin.ExternalPluginSpec
 	LogFile             string
 	// LogLevel can be "warn", "info", "debug", or "trace".
-	LogLevel string
+	LogLevel            string
```
Comment (ekinanp) [inline cmd/internal/server/core.go]: I think we should include the `Socket` as an option, or make it a parameter to `New`. That would remove the config dependency and make the `Server` object more generic.

Comment (MikaelSmith) [reply cmd/internal/server/core.go]: That makes sense.

**Comment by MikaelSmith** [comment]
I think we should also roll plugins.yaml into this, as noted in #132.

File: cmd/server.go [hunk 2]
```diff
@@ -62,24 +72,48 @@ func serverMain(cmd *cobra.Command, args []string) exitCode {
...
+	if err != nil {
+		panic(err.Error())
+	}
+	if err := config.ReadFrom(configFile); err != nil {
+		return server.Opts{}, err
+	}
+
+	// Unmarshal the external plugins, if any are specified
+	var externalPlugins []plugin.ExternalPluginSpec
+	if externalPluginsRaw := viper.Get("external_plugins"); externalPluginsRaw != nil {
```
Comment (MikaelSmith) [inline cmd/server.go]: Why is this underscore, unlike all the other flags?

Comment (ekinanp) [reply cmd/server.go]: Oh, that's not a flag anymore. It's underscore b/c in the YAML, it'd be specified under an `external_plugins` key. Don't mind changing it to `external-plugins` though

Comment (MikaelSmith) [reply cmd/server.go]: Can you right your YAML file without quoting it that way?

**Pull request merged**